### PR TITLE
Nazar/feature/product prep routes

### PR DESCRIPTION
**Pull request by Nazar-kob**

### CODE REVIEW

File: backend/app/router/utils.py [hunk 1]
```diff
@@ -37,3 +37,17 @@ def check_access_to_product(product: m.Product, user: m.User, product_id: int) -
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND, detail="Product was not found"
         )
+
+
+def check_access_to_product_prep(prep_id: int, product: m.Product, prep: m.Prep):
```
[the remark this answers is not in the document]
Comment (Nazar-kob) [reply backend/app/router/utils.py:42]: sometimes prep can be Nole so  we can not get prep.id

File: backend/app/model/product.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ class Product(Base):
     business_id = Column(Integer, ForeignKey("businesses.id"))
     name = Column(String(128), nullable=False)
     price = Column(Float, default=0.0)
-    sold_by = Column(Enum(SoldBy), default=SoldBy.unknown)
+    sold_by = Column(Enum(SoldBy), default=SoldBy.unknown)  # TODO add both
```
[the remark this answers is not in the document]
Comment (Nazar-kob) [reply backend/app/model/product.py:17]: not yet

File: backend/app/schema/prep_schemas.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        orm_mode = True
+
+
+class ProductPrepsOut(BaseModel):
+    id: int
+    preps: list[ProductPrepOut]
+
+
+class UpdateProductPrep(BaseModel):
+    is_active: bool
```
[the remark this answers is not in the document]
Comment (Nazar-kob) [reply backend/app/schema/prep_schemas.py:25]: I use black I don't know why!

File: backend/app/router/product.py [hunk 4]
```diff
@@ -95,3 +99,95 @@ def update_product(
     db.refresh(product)
 
     return product
+
+
+@router.post(
+    "/{id}/prep",
+    status_code=status.HTTP_201_CREATED,
```
Comment (kipod) [inline backend/app/router/product.py:106]: response_model=s. ProductPrepOut

File: backend/app/router/product.py [hunk 5]
```diff
@@ -95,3 +99,95 @@ def update_product(
...
+    db.add(prep)
+    db.commit()
+    db.refresh(prep)
+
+    return prep
+
+
+@router.get(
+    "/{id}/prep",
+    status_code=status.HTTP_200_OK,
```
Comment (kipod) [inline backend/app/router/product.py:130]: response_model=s. ProductPrepsOut

**Review by kipod — APPROVED** [review]
Super. Thanks

**Pull request merged**